ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderElection.java

```diff
@@ -273,6 +275,10 @@ private ResultAndTerm submitRequestAndWaitResult(Phase phase, RaftConfigurationI
       throws InterruptedException {
     final ResultAndTerm r;
     final Collection<RaftPeer> others = conf.getOtherPeers(server.getId());
+    if (!conf.containsInBothConfs(server.getId())) {
+      r = new ResultAndTerm(Result.REJECTED, electionTerm);
```

Comment:
Let's add a new Result.NOT_IN_CONF but not using Result.REJECTED.  Also, the if-statement should be moved up.
```
@@ -104,7 +105,7 @@ class LeaderElection implements Runnable {
     ELECTION
   }
 
-  enum Result {PASSED, REJECTED, TIMEOUT, DISCOVERED_A_NEW_TERM, SHUTDOWN}
+  enum Result {PASSED, REJECTED, TIMEOUT, DISCOVERED_A_NEW_TERM, SHUTDOWN, NOT_IN_CONF}
 
   private static class ResultAndTerm {
     private final Result result;
@@ -271,6 +272,9 @@ class LeaderElection implements Runnable {
 
   private ResultAndTerm submitRequestAndWaitResult(Phase phase, RaftConfigurationImpl conf, long electionTerm)
       throws InterruptedException {
+    if (!conf.containsInConf(server.getId())) {
+      return new ResultAndTerm(Result.NOT_IN_CONF, electionTerm);
+    }
     final ResultAndTerm r;
     final Collection<RaftPeer> others = conf.getOtherPeers(server.getId());
     if (others.isEmpty()) {
@@ -322,6 +326,7 @@ class LeaderElection implements Runnable {
             continue; // should retry
           case REJECTED:
           case DISCOVERED_A_NEW_TERM:
+          case NOT_IN_CONF:
             final long term = r.maxTerm(server.getState().getCurrentTerm());
             server.changeToFollowerAndPersistMetadata(term, r);
             return false;
```